version: "3"

services:
  apisix:
    image: apache/apisix:3.8.0-debian
    restart: always
    volumes:
      - ./apisix_log:/usr/local/apisix/logs
      - ./apisix_conf/config.yaml:/usr/local/apisix/conf/config.yaml:ro
      - ./apisix_conf/debug.yaml:/usr/local/apisix/conf/debug.yaml:ro
    ports:
      - "9080:9080/tcp"
      - "9091:9091/tcp"
      - "9443:9443/tcp"
      - "9180:9180/tcp"
    networks:
      - apisix
    depends_on:
      - etcd
    extra_hosts:
      - "nutrix-hub-api:host-gateway"
      - "zitadel:host-gateway"

  apisix-dashboard:
    image: apache/apisix-dashboard:3.0.1-alpine
    restart: always
    volumes:
      - ./dashboard_conf/conf.yaml:/usr/local/apisix-dashboard/conf/conf.yaml
    ports:
      - "9000:9000/tcp"
    networks:
      - apisix
    depends_on:
      - apisix
    extra_hosts:
      - "nutrix-hub-api:host-gateway"


  etcd:
    image: quay.io/coreos/etcd:v3.5.9  # Using official etcd image
    container_name: etcd
    volumes:
      - ./etcd_data:/etcd-data
    environment:
      ETCD_ENABLE_V2: "true"
      ETCD_LISTEN_CLIENT_URLS: "http://0.0.0.0:2379"
      ETCD_ADVERTISE_CLIENT_URLS: "http://etcd:2379"
      ETCDCTL_API: "3"
      ETCD_DATA_DIR: "/etcd-data"
    ports:
      - "2379:2379"
    healthcheck:
      test: ["CMD", "etcdctl", "--endpoints=http://localhost:2379", "endpoint", "health"]
      interval: 5s
      timeout: 2s
      retries: 10
    networks:
      - apisix

networks:
  apisix:
    driver: bridge
